### ISSUE TITLE
Missing files for 2021.1

### ISSUE THREAD

**Issue by magnifikus** [z4yx/petalinux-docker#12]
for 2021.1:

update to ubuntu 20.4
add xxd libtinfo5  

still trying to get a successfull build, will report

**Comment by z4yx** [comment]
I have verified that Petalinux 2021.1 works fine with current Dockerfile.

```
Step 11/20 : COPY accept-eula.sh ${PETA_RUN_FILE} /
 ---> 78733c580127
Step 12/20 : RUN chmod a+rx /${PETA_RUN_FILE} &&   chmod a+rx /accept-eula.sh &&   mkdir -p /opt/Xilinx &&   chmod 777 /tmp /opt/Xilinx &&   cd /tmp &&   sudo -u vivado -i /accept-eula.sh /${PETA_RUN_FILE} /opt/Xilinx/petalinux &&   rm -f /${PETA_RUN_FILE} /accept-eula.sh
 ---> Running in b99a9861e648
spawn /petalinux-v2021.1-final-installer.run /opt/Xilinx/petalinux
ERROR: Invalid options: /opt/Xilinx/petalinux
spawn /petalinux-v2021.1-final-installer.run -d /opt/Xilinx/petalinux
INFO: Checking installation environment requirements...
INFO: Checking free disk space
INFO: Checking installed tools
INFO: Checking installed development libraries
INFO: Checking network and other services
WARNING: No tftp server found - please refer to "UG1144  PetaLinux Tools Documentation Reference Guide" for its impact and solution
INFO: Checking installer checksum...
INFO: Extracting PetaLinux installer...

LICENSE AGREEMENTS

PetaLinux SDK contains software from a number of sources.  Please review
the following licenses and indicate your acceptance of each to continue.

You do not have to accept the licenses, however if you do not then you may 
not use PetaLinux SDK.

Use PgUp/PgDn to navigate the license viewer, and press 'q' to close

Press Enter to display the license agreements
WARNING: terminal is not fully functional
Do you accept Xilinx End User License Agreement? [y/N] > y
WARNING: terminal is not fully functional
Do you accept Webtalk Terms and Conditions? [y/N] > y)
WARNING: terminal is not fully functional
Do you accept Third Party End User License Agreement? [y/N] > yment.txt  (press RETURN)
INFO: Installing PetaLinux...
INFO: Checking PetaLinux installer integrity...
INFO: Installing PetaLinux SDK to "/opt/Xilinx/petalinux/."
INFO: Installing buildtools in /opt/Xilinx/petalinux/./components/yocto/buildtools
INFO: Installing buildtools-extended in /opt/Xilinx/petalinux/./components/yocto/buildtools_extended
INFO: PetaLinux SDK has been installed to /opt/Xilinx/petalinux/.
Removing intermediate container b99a9861e648
 ---> d8b229fa6c00
Step 13/20 : RUN echo "dash dash/sh boolean false" | debconf-set-selections
 ---> Running in 8dd18c046572
Removing intermediate container 8dd18c046572
 ---> b82250802290
Step 14/20 : RUN DEBIAN_FRONTEND=noninteractive dpkg-reconfigure dash
 ---> Running in 7fc1da51b1bd
Removing 'diversion of /bin/sh to /bin/sh.distrib by dash'
Adding 'diversion of /bin/sh to /bin/sh.distrib by bash'                                           
Removing 'diversion of /usr/share/man/man1/sh.1.gz to /usr/share/man/man1/sh.distrib.1.gz by dash' 
Adding 'diversion of /usr/share/man/man1/sh.1.gz to /usr/share/man/man1/sh.distrib.1.gz by bash'   
Removing intermediate container 7fc1da51b1bd                                                       
 ---> 60b827c3d913
Step 15/20 : USER vivado
 ---> Running in 693728eb0a08
Removing intermediate container 693728eb0a08
 ---> b5de7e477f58
Step 16/20 : ENV HOME /home/vivado
 ---> Running in b49e63c1054d
Removing intermediate container b49e63c1054d
 ---> 28e8b536cf80
Step 17/20 : ENV LANG en_US.UTF-8
 ---> Running in 7572559be5c3
Removing intermediate container 7572559be5c3
 ---> 77d38841c3f2
Step 18/20 : RUN mkdir /home/vivado/project
 ---> Running in 7ff339d9379e
Removing intermediate container 7ff339d9379e
 ---> 07c7f75da634
Step 19/20 : WORKDIR /home/vivado/project
 ---> Running in b7a3dd8de352
Removing intermediate container b7a3dd8de352
 ---> 28b154262f58
Step 20/20 : RUN echo "source /opt/Xilinx/petalinux/settings.sh" >> /home/vivado/.bashrc
 ---> Running in 8f38439c1d6b
Removing intermediate container 8f38439c1d6b
 ---> f1514b9c3384
Successfully built f1514b9c3384
Successfully tagged petalinux:2021.1

```

**Comment by magnifikus** [comment]
Installing works but building not

**Comment by z4yx** [comment]
Could you provide a sample hdf file so I can test.

**Comment by magnifikus** [comment]
Sorry from mobile all the time, happens with official kria 260 bsp. Will provide a log later.

**Comment by magnifikus** [comment]
build:
```docker build --build-arg PETA_VERSION=2021.1 --build-arg PETA_RUN_FILE=petalinux-v2021.1-final-installer.run -t petalinux:2021.1 .```

create project:
```
vivado@d3febeb86dcd:~/project$ petalinux-create -t project -s xilinx-k26-som-v2021.1-updated-final.bsp -n testprj
INFO: Create project: testprj
INFO: New project successfully created in /home/vivado/project/testprj
```

build
```
vivado@d3febeb86dcd:~/project/testprj$ petalinux-build
[INFO] Sourcing buildtools
[INFO] Building project
[INFO] Generating Kconfig for project
[INFO] Silentconfig project
[INFO] Extracting yocto SDK to components/yocto. This may take time!
[INFO] Sourcing build environment
[INFO] Generating kconfig for Rootfs
[INFO] Silentconfig rootfs
[INFO] Generating plnxtool conf
[INFO] Adding user layers
ERROR: Failed to add user layer: /home/vivado/project/testprj/project-spec/meta-user
ERROR: Failed to build project. Check the /home/vivado/project/testprj/build/build.log file for more details...
```

content of  build/build.log
```
[INFO] Adding user layers
NOTE: Starting bitbake server...
NOTE: No reply from server in 30s
ERROR: The following required tools (as specified by HOSTTOOLS) appear to be unavailable in PATH, please install them in order to proceed:
  xxd
ERROR: The following required tools (as specified by HOSTTOOLS) appear to be unavailable in PATH, please install them in order to proceed:
  xxd
ERROR: Exception in server main event loop running command [] ()
Traceback (most recent call last):
  File "/home/vivado/project/testprj/components/yocto/layers/core/bitbake/lib/bb/server/process.py", line 217, in ProcessServer.main():
                         serverlog("Running command %s" % command)
    >                    self.command_channel_reply.send(self.cooker.command.runCommand(command))
                         serverlog("Command Completed")
  File "/home/vivado/project/testprj/components/yocto/layers/core/bitbake/lib/bb/command.py", line 67, in Command.runCommand(commandline=[], ro_only=False):
             if command != "updateConfig" and command != "setFeatures":
    >            self.cooker.init_configdata()
                 if not self.remotedatastores:
  File "/home/vivado/project/testprj/components/yocto/layers/core/bitbake/lib/bb/cooker.py", line 235, in BBCooker.init_configdata():
             if not hasattr(self, "data"):
    >            self.initConfigurationData()
                 bb.debug(1, "BBCooker parsed base configuration %s" % time.time())
  File "/home/vivado/project/testprj/components/yocto/layers/core/bitbake/lib/bb/cooker.py", line 360, in BBCooker.initConfigurationData():
             self.databuilder = bb.cookerdata.CookerDataBuilder(self.configuration, False)
    >        self.databuilder.parseBaseConfiguration()
             self.data = self.databuilder.data
  File "/home/vivado/project/testprj/components/yocto/layers/core/bitbake/lib/bb/cookerdata.py", line 303, in CookerDataBuilder.parseBaseConfiguration():
             except (SyntaxError, bb.BBHandledException):
    >            raise bb.BBHandledException()
             except bb.data_smart.ExpansionError as e:
bb.BBHandledException

Timeout while waiting for a reply from the bitbake server (60s)
ERROR: Failed to build project. Check the /home/vivado/project/testprj/build/build.log file for more details...
```

fix:
```
sudo apt-get update && sudo apt-get install xxd

....
[INFO] Sourcing buildtools
[INFO] Building project
[INFO] Sourcing build environment
[INFO] Adding user layers
[INFO] Generating workspace directory
INFO: bitbake petalinux-image-minimal
NOTE: Started PRServer with DBfile: /home/vivado/project/testprj/build/cache/prserv.sqlite3, IP: 127.0.0.1, PORT: 40905, PID: 3325
WARNING: You are running bitbake under WSLv2, this works properly but you should optimize your VHDX file eventually to avoid running out of storage space
Loading cache: 100% |                                                                                                                                                                                                        | ETA:  --:--:--
Loaded 0 entries from dependency cache.
...
```


i used 20.04 base image and had also an error related to this: https://github.com/Xilinx/meta-xilinx-tools/issues/19

fix:
```
sudo apt-get update && sudo apt-get install libtinfo5
```

its possible that the 2nd fix is only necessary for 20.04!

**Comment by z4yx** [comment]
Thank you for providing the logs. I've added these 2 packages to the installation step.
